components/brave_wallet_ui/common/async/wallet_async_handler.ts

```diff
@@ -93,6 +95,17 @@ async function refreshWalletInfo (store: Store) {
     prices: getTokenPrices
   }
   store.dispatch(WalletActions.tokenBalancesUpdated(tokenBalancesAndPrices))
+
+  // Update Portfolio Price History
+  const getTokenPriceHistory = await Promise.all(state.accounts.map(async (account) => {
+    return Promise.all(account.tokens.map(async (token) => {
+      return {
+        token: token,
+        history: await assetPriceController.getPriceHistory(token.asset.symbol.toLowerCase(), state.selectedPortfolioTimeline)
+      }
+    }))
+  }))
```

Comment:
I have a couple of remarks here:
* `token.asset.symbol` is not unique for ERC-20 tokens. Although not very common, it's possible for two tokens to have the same `symbol`. Can we instead use contract address for fetching pricing information? @bbondy 
* It makes sense to optimize this function, as it'll have a direct impact on the loading of the main wallet dashboard.

    Here's one idea:
    ```js
    const tokens = [
      ...new Set(state.accounts.flatMap(account => account.tokens).map(token => token.symbol))
    ]
    ```

    Then perform `tokens.reduce` to build the following object:
    ```js
    {
       DAI: <history>,
       USDC: <history>
    }
    ```

    So whenever you need the history of a particular token, it'll be an `O(1)` lookup.

What do you think?

Reply:
- I agree with fetching price informations using a `contract address`. Currently I'm not sure if we are able to since we are getting information from `CoinGecko` Will need to investigate if they have a param for `contract address` in their api

- For this optimization, while in theory would help with load times. I think it could effect a couple things.

1. If the user loads the wallet page and then comes back 40 minutes later, the history is then going to be outdated when they click to view a tokens history.
2. The user also has the ability to change the `TimeFrame` which is a param in our `api call` by default its `1 day`. So every-time they change, it's fetching new history and would make the list of `Price Histories` the wrong `TimeFrame` unless we refetched price history for all of the users tokens when they made that action, but i think that would make load times for clicking on a single asset longer.
3. The Portfolio `TimeFrame` and Selected Asset `TimeFrame` are able to be different. For instance, i could be on the Portfolio page and have the `TimeFrame` set to `7 days` and then select an asset and it's `TimeFrame` can be set to `1 day`